src/EbmlDate.cpp

```diff
@@ -26,8 +26,8 @@ filepos_t EbmlDate::ReadData(IOCallback & input, ScopeMode ReadFully)
   binary Buffer[8];
   input.readFully(Buffer, GetSize());
 
-  Value = endian::from_big64(Buffer);
-  SetValueIsSet();
+  double NewValue = endian::from_big64(Buffer);
```

Comment:
This looks like a copy-paste error. `EbmlDate` is based on an `EbmlSInteger`. There's no place for a `double` here. One of my tests failing confirms that this is actually an issue; the test case reads the segment info, modifies the title & writes the whole segment info element back to the file. It does not touch the `KaxDate` element at all. Yet its value is different after writing it back, meaning that libEBML messes the value up.

If I replace this line with `int64_t NewValue = …` my test case passes.

Reply:
Indeed, good catch! Fixed in the new version.